app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -187,7 +188,7 @@ public void reportProgress(int progress) {
                         if (ukr == null) {
                             Logger.log(AndroidLogger.TYPE_ERROR_ASSERTION, "Shouldn't be able to not have a valid key record when OTA restoring with a key server");
                             this.publishProgress(PROGRESS_DONE);
-                            return PullTaskResult.UNKNOWN_FAILURE;
+                            return new Pair<>(PullTaskResult.UNKNOWN_FAILURE, "Unable to generate encryption key");

```

Comment:
Not going to localize this error message or the other 2 above because I don't think they really ever happen in practice.